rust/src/nfs/nfs.rs

```diff
@@ -1082,15 +1207,17 @@ impl NFSState {
                         // but lower than the record size
                         let n1 = cmp::max(cur_i.len(), 1024);
                         let n2 = cmp::min(n1, rec_size);
-                        return AppLayerResult::incomplete((i.len() - cur_i.len()) as u32, n2 as u32);
+                        return AppLayerResult::incomplete(stream_slice.len() - cur_i.len() as u32, n2 as u32);
                     }
 
                     // we have the full records size worth of data,
                     // let's parse it
                     match parse_rpc(&cur_i[..rec_size]) {
                         Ok((_, ref rpc_record)) => {
+                            self.add_rpc_tcp_ts_creds(flow, stream_slice, cur_i, (cur_i.len() - rpc_record.prog_data.len()) as i64);
```

Comment:
correct, this needs some cleanups, Is it a good idea to make a vairable inside `add_rcp_ts_creds` to do the length calculation?

Reply:
maybe easier to store the creds slice or length in the `rpc_record` and use that?

Reply:
I guess! I'll try this out.

Reply:
alright, since what the `nfs4.creds` care about is the `machine_name` `machine_len` of an `AUTH_UNIX` creds, 
I think I could follow the logic of checking `if creds_flavor == AUTH_UNIX` and then return the needed fields, or might as well return the `creds_buf` as a whole

Reply:
not sure I understand the machine name comment. A `rpc.creds` frame should contain all parts of the creds fields. I'd think the len, flavor and then the actual data. Rule writers can then use the flavor field to check its unix, and then find the machine name easily?

Reply:
Yes, okay good. on it.